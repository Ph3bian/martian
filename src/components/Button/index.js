import React from "react";
// import PropTypes from "prop-types";
import styles from "./button.module.scss";

const Button = ({ type, children, text, onClick, variant, ...rest }) => (
  <button
    type={type || "button"}
    onClick={onClick}
    className={[styles.Button, styles[variant]].join(" ")}
    {...rest}
  >
    {text || children}
  </button>
);

// Button.propTypes = {
//   variant: PropTypes.string,
//   text: PropTypes.string,
//   type: PropTypes.string.isRequired,
//   onClick: PropTypes.func,
// };

// Button.defaultProps = {
//   variant: "",
//   text: "Submit",
//   onClick: () => null,
// };

export default Button;
