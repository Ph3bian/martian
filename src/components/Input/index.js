import React from "react";
import PropTypes from "prop-types";
import styles from "./input.module.scss";

const Input = ({ label, error, ...rest }) => (
  <div className={styles.Input}>
    <label>{label}</label>
    <div className={styles.InputField}>
      <input {...rest} className={error && "error"} />
      {error && (
        <div>
          <span className={styles.InputError}> {error}</span>
        </div>
      )}
    </div>
  </div>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  rest: PropTypes.shape({}),
};

Input.defaultProps = {
  error: "",
  rest: null,
};
export default Input;
