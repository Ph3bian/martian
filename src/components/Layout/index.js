import React, { useEffect } from "react";
import { Logo } from "assets/svg";
import styles from "./layout.module.scss";
import {handleTheme} from "./functions"

const Layout = ({ children }) => {
  useEffect(() => {
    if (localStorage.getItem("theme")) {
      document.documentElement.setAttribute("data-theme",  localStorage.getItem("theme"));
    }
  }, []);
  return (
    <div className={styles.Layout}>
      <div className={styles.LayoutHeader}>
       Martian
        <div onClick={handleTheme}> click</div>
      </div>
      <div className={styles.LayoutBody}>{children}</div>
    </div>
  );
};
export default Layout;
