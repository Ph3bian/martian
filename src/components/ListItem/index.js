import React, { Fragment } from "react";
import styles from "./listItem.module.scss";
const ListItem = ({ handleClick, data }) => {
  return (
    <Fragment>
      {data &&
        data.map((item) => (
          <div className={styles.ListItem}>
            <div className={styles.ListItemDescription}>
              <h2>{item.name}</h2>
              <p>{item.budget}</p>
              <span>{item.budget_spent}</span>
            </div>
            <div className={styles.ListItemArrow}>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
                <path d="M0 0h24v24H0z" fill="none" />
                <path d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z" />
              </svg>
            </div>
          </div>
        ))}
    </Fragment>
  );
};
export default ListItem;
