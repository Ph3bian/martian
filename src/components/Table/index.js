import React from "react";
import styles from "./table.module.scss";
const Table = ({ data = [], headers = [], handleClick }) => {
  const handleRowClick = (item) => {
    return handleClick && handleClick(item);
  };
  return (
    <table className={styles.Table}>
      <thead>
        <tr>
          {headers.map(({ name, title }, id) => (
            <th key={`${name}-${id}`}>{title}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {data.map((item, id) => (
          <tr onClick={() => handleRowClick(item)} key={id}>
            {headers.map((header, id) => (
              <td key={`${header.name}-${id}`}>
                {header.formatter ? header.formatter(item) : item[header.name]}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default Table;
