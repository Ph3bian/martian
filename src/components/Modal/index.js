import React, { useRef, useCallback, useEffect } from "react";
import styles from "./modal.module.scss";
import { CloseIcon } from "../../assets/svg";
import Button from "../Button";

const Modal = ({
  children,
  title,
  setShown,
  shown,
  hasFooter,
  hasHeader = true,
  handleSubmit,
  loading,
}) => {

  const node = useRef(null);

  const handleHide = useCallback(
    (event) => {
      if (event.key === "Escape") {
        setShown(!shown);
        return;
      }
    },
    [setShown, shown]
  );
  const handleClick = useCallback(
    (event ) => {
      if (node.current && !node.current.contains( event.target)) {
        setShown(!shown);
        return;
      }
    },
    [setShown, shown]
  );

  useEffect(() => {
    document.addEventListener("keydown", handleHide);
    document.addEventListener("click", handleClick);

    return () => {
      document.removeEventListener("keydown", handleHide);
      document.removeEventListener("click", handleClick);
    };
  }, [handleClick, handleHide]);

  return (
    <div tabIndex="-1" role="dialog" className={styles.Modal}  aria-labelledby="modalTitle" aria-describedby="modalDesc" data-show={shown}>
      <div className={styles.ModalContent}  ref={node}>
        {hasHeader && (
          <div className={styles.header}>
            <h3 id="modalTitle">{title}</h3>
            <CloseIcon
             aria-label="Close"
              role="button"
              onKeyDown={() => setShown(!shown)}
              onClick={() => setShown(!shown)}
            />
          </div>
        )}
        <div className={styles.body}>
          <div className={styles.bodyContent} id="modalDesc">{children}</div>
        </div>
        {hasFooter && (
          <div className={styles.footer}>
            <Button
              type="button"
              variant="neutral"
              onKeyDown={() => setShown(!shown)}
              onClick={() => setShown(!shown)}
            >
              Close
            </Button>
            <Button
              type="button"
              data-dismiss="modal"
              variant="primary"
              onClick={handleSubmit}
              disabled={loading}
              name="submit"
            >
              {loading ? "loading..." : "Submit"}
            </Button>
          </div>
        )}
      </div>
    </div>
  );
};

export default Modal;
