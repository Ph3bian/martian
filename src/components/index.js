import Button from "./Button";
import Input from "./Input";
import Table from "./Table";
import ListItem from "./ListItem";
import Layout from "./Layout";
import Modal from "./Modal";

export { Button, Input, Layout, Table, Modal, ListItem };
