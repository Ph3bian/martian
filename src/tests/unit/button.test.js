import React from 'react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { render, fireEvent } from '@testing-library/react';
import {Button} from 'components';

describe('The <Button/> component', () => {
  expect.extend(toHaveNoViolations);

  const defaultProps = {
    type: 'button',
    onClick: jest.fn(),
    text: 'Submit',
  };

  test('Should render button correctly', () => {
    const { asFragment } = render(<Button {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render button text correctly', () => {
    const { getByText } = render(<Button {...defaultProps} />);
    expect(getByText(/Submit/)).toBeInTheDocument();
  });

  test('Should call the onClick handler when it is provided', () => {
    const { getByText } = render(<Button {...defaultProps} />);
    fireEvent.click(getByText(defaultProps.text));
    expect(defaultProps.onClick).toHaveBeenCalled();
  });

  test('Should not fail any accessibility tests', async () => {
    const { container } = render(<Button {...defaultProps} />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
