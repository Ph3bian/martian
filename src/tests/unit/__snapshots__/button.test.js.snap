// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Button/> component Should render button correctly 1`] = `
<DocumentFragment>
  <button
    class="Button undefined"
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;
